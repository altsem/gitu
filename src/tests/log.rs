use super::*;

fn setup() -> TestContext {
    let ctx = TestContext::setup_clone();
    commit(ctx.dir.path(), "this-should-be-at-the-top", "");
    commit(ctx.dir.path(), "this-should-not-be-visible", "");
    ctx
}

#[test]
fn log_n_prompt_show() {
    snapshot!(setup(), "l-n-n");
}

#[test]
fn log_n_prompt_valid() {
    snapshot!(setup(), "l-n-n10<enter>");
}

#[test]
fn log_n_prompt_invalid() {
    snapshot!(setup(), "l-n-nfff<enter>");
}

#[test]
fn log_grep_prompt_show() {
    snapshot!(setup(), "l-F");
}

#[test]
fn log_grep_prompt_valid() {
    snapshot!(setup(), "l-F");
}

#[test]
fn log_grep_prompt_invalid() {
    snapshot!(setup(), "l-Fui<enter>");
}

#[test]
fn log_other_prompt() {
    snapshot!(setup(), "lljlo");
}

#[test]
fn log_other() {
    snapshot!(setup(), "lljlo<enter>");
}

#[test]
fn log_other_input() {
    snapshot!(setup(), "lomain~1<enter>");
}

#[test]
fn log_other_invalid() {
    snapshot!(setup(), "lo <enter>");
}
