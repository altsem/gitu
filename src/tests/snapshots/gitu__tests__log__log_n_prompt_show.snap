---
source: src/tests/log.rs
expression: ctx.redact_buffer()
---
▌On branch main                                                                 |
▌Your branch is ahead of 'origin/main' by 2 commit.                             |
                                                                                |
 Recent commits                                                                 |
 _______ main add this-should-not-be-visible                                    |
 _______ add this-should-be-at-the-top                                          |
 _______ origin/main add initial-file                                           |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
────────────────────────────────────────────────────────────────────────────────|
? Limit number of commits (default 256): ›                                      |
────────────────────────────────────────────────────────────────────────────────|
Log                     Arguments                                               |
l Log current           -F Search messages (--grep)                             |
o Log other             -n Limit number of commits (-n)                         |
q/<esc> Quit/Close                                                              |
styles_hash: 53b6b1f36c36ef42
